### PR TITLE
Support for new values in boost feature

### PR DESCRIPTION
**Pull request by Doc94**
**Description:** Well discord release the boost feature for guilds, and add new values in Guild properties 
https://discordapp.com/developers/docs/resources/guild#guild-object-premium-tier
https://discordapp.com/developers/docs/resources/guild

**Justification:** Support to boost features (read values)

### CODE REVIEW

**Comment by Doc94** [comment]
My confusion its how to name "this" in docs, because the public name its "boost" but discord docs show "premium" or "subscription"

**Comment by SizableShrimp** [comment]
I believe things are generally documented how discord docs also name them, e.g. how in the library and discord docs it is called Guild not server like it is in the client

File: core/src/main/java/discord4j/core/object/data/stored/BaseGuildBean.java [hunk 1]
```diff
@@ -269,6 +276,8 @@ public void setVerificationLevel(final int verificationLevel) {
         this.verificationLevel = verificationLevel;
     }
 
+    public int getPremiumTier() { return premiumTier; }
```
Comment (quanticc) [inline core/src/main/java/discord4j/core/object/data/stored/BaseGuildBean.java]: Needs reformatting. We use this code style for the project https://github.com/Discord4J/meta/blob/master/Discord4J.xml

Happens a couple of times below too.

Comment (Doc94) [reply core/src/main/java/discord4j/core/object/data/stored/BaseGuildBean.java]: fixed in #b8949e2

**Pull request merged**